app/models/remote_authenticatable.rb

```diff
@@ -72,35 +72,34 @@ def create_challenge_flow(resp)
         self
       end
 
-      def auth_complete(resp)
-        access_token = resp[:authentication_result][:access_token]
-        aws_user = get_user_info(access_token)
-        user_attributes = get_user_attributes(aws_user)
-        self.login_id = user_attributes['email']
-        self.user_id = aws_user.username
-        self.email = user_attributes['email']
-        self.phone_number = user_attributes['phone_number']
-        self.access_token = access_token
-        self.roles = user_attributes['custom:roles']
-        self.permissions = UserRolePermissions.new(user_attributes['custom:roles'], user_attributes['email'])
-        self.full_name = "#{user_attributes['given_name']} #{user_attributes['family_name']}"
-        self.given_name = user_attributes['given_name']
-        self.family_name = user_attributes['family_name']
-        self.mfa = aws_user.preferred_mfa_setting
-        # session_start_time will be serialised to a string so
-        # lets be explicit about setting it as a string here.
+      def auth_complete(resp, params)
+        claims = get_user_info(resp[:authentication_result][:id_token])
+        self.login_id = claims['email']
+        self.user_id = claims['sub']
+        self.email = claims['email']
+        self.phone_number = claims['phone_number']
+        self.access_token = resp[:authentication_result][:access_token]
+        self.roles = claims['custom:roles']
+        self.permissions = UserRolePermissions.new(claims['custom:roles'], claims['email'])
+        self.full_name = "#{claims['given_name']} #{claims['family_name']}"
+        self.given_name = claims['given_name']
+        self.family_name = claims['family_name']
+        self.team = Team.find_by_team_alias(claims['cognito:groups'][0])&.id
+        self.cognito_groups = claims['cognito:groups']
+        self.mfa = claims['mfa'] || params[:challenge_name]
         self.session_start_time = Time.now.to_s
         self
       end
 
-      def get_user_info(access_token)
-        SelfService.service(:cognito_client).get_user(access_token: access_token)
+      def get_user_info(jwt)
+        header = JSON.parse(Base64.decode64(jwt.split('.')[0]))
+        jwk = find_jwk(header['kid'])
+        JSON::JWT.decode(jwt, jwk)
```

Comment:
We really should be supplying the algorithm here that we wish to use to verify the JWT as it's possible that we could be supplied a [fraudulent JWT containing a 'none' algorithm](https://auth0.com/blog/critical-vulnerabilities-in-json-web-token-libraries/#Meet-the--None--Algorithm). It looks like the JWT library supports [it](https://github.com/nov/json-jwt/blob/master/lib/json/jose.rb#L59) but I'd like to see some tests to confirm it.

Reply:
Moved from autodetect which does from the jwk to specifying the algorithm.  So the line now looks like...

`JSON::JWT.decode(jwt, jwk, :RS256)`